specs/IsSmartScreenRequired

```diff
@@ -0,0 +1,94 @@
+
+Instead we have CoreWebView2Settings.IsSmartScreenRequired. Each WebView2 declares if it requires SmartScreen. Some WebView2s may be used to display app content and don't require SmartScreen and others may be rendering arbitrary web content and do need SmartScreen. Having SmartScreen on unnecessarily for app content is a detriment to performance but otherwise not a problem. Having SmartScreen off for arbitrary web content is an issue. We have to turn SmartScreen on or off for all the WebView2s in an environment so if any WebView2 requires SmartScreen then we turn it on for all of them. If WebView2 settings change or WebView2s are closed and then all the WebView2s in an environment don't require SmartScreen, then we can turn SmartScreen off.
+
+It is much easier to indicate if individual WebView2s require SmartScreen than to have an end developer explicitly manage if SmartScreen should be enabled as a whole, especially when its different sets of WebView2s in different processes (like Excel's set of WebView2s and Word's set of WebView2s) all sharing the same user data folder.
+In this document we describe the new setting. We'd appreciate your feedback.
+
+
+# Description
+You can use CoreWebView2Settings.IsSmartScreen to control SmartScreen. SmartScreen is enabled or disabled per browser process, so all WebView2 applications sharing the same user data folder path also share SmartScreen being enabled or disabled.
+If CoreWebView2Setting.IsSmartScreenRequired is true for any CoreWebView2 in its associated CoreWebView2Environment, then SmartScreen is enabled. If CoreWebView2Setting.IsSmartScreenRequired is false for all CoreWebView2s in their CoreWebView2Environment, then SmartScreen is disabled.
+The default value for `IsSmartScreenRequired` is true if SmartScreen is enabled when the CoreWebView2 is created and false if SmartScreen is disabled when the CoreWebView2 is created. The value doesn't change if SmartScreen is enabled or disabled later.
+
+Changes to `IsSmartScreenRequired` take effect on the next navigation or download.
+
+
+# Examples
+```cpp
+wil::com_ptr<ICoreWebView2Settings> m_webViewSettings;
+void SettingsComponent::ToggleSmartScreenRequired()
+{
+    wil::com_ptr<ICoreWebView2Settings11> coreWebView2Settings11;
+    coreWebView2Settings11 =
+                m_webViewSettings.try_query<ICoreWebView2Settings11>();
+    if(coreWebView2Settings11) 
+    {
+        BOOL enabled;
+        CHECK_FAILURE(coreWebView2Settings11->get_IsSmartScreenRequired(&enabled));
+        CHECK_FAILURE(coreWebView2Settings11->put_IsSmartScreenRequired(enabled ? FALSE : TRUE));
+    }
+}
+```
+
+```c#
+void ToggleSmartScreenRequired()
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsSmartScreenRequired = !settings.IsSmartScreenRequired;
+}
+```
+
+# Remarks
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
```

Comment:
I assume that if I set `IsSmartScreenRequired = true`, but also `options->put_AdditionalBrowserArguments(L"--disable-features=msSmartScreenProtection")`, then the browser argument overrides the CoreWebView2 setting. (We have to do it this way for compatibility, I assume.)

Reply:
Do we document the `--disable-features=msSmartScreenProtection` command line option anywhere already? If we already document it, then we should additionally document here how the two interact. If we don't already document it, then we don't need to talk about it in the documentation here.